src/lib.rs

```diff
@@ -146,10 +162,18 @@ pub trait Reserve {
 	fn reserve(&mut self, additional: usize);
 }
 
+/// Lockable collection that can extend their capacity.
+pub trait ReserveLock {
+	/// Reserve enough memory for `additional` more elements.
+	///
+	/// It is expected that this method may block.
+	fn reserve_lock(&self, additional: usize);
+}
+
 /// Queryable collection.
 pub trait Get<T>: Collection {
 	/// Returns a reference to the item stored behind the given key (if any).
-	fn get(&self, key: T) -> Option<&Self::Item>;
+	fn get<'a>(&'a self, key: T) -> Option<Self::ItemRef<'a>>;
```

Comment:
You don't need to use an explicit lifetime here (and other similar functions) and instead use the placeholder lifetime `'_` to make it less verbose:
```rust
fn get(&self, key: T) -> Option<Self::ItemRef<'_>>;
```